backend/src/main/java/com/bulletjournal/messaging/MessagingService.java

```diff
@@ -283,8 +288,7 @@ public MailjetEmailParams createEmailPramsForAppInvitation(String inviter,
     }
 
     private boolean isValidEmailAddr(String email) {
-        String emailPattern = "^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,}$";
-        Matcher emailMatcher =  Pattern.compile(emailPattern, Pattern.CASE_INSENSITIVE)
+        Matcher emailMatcher =  Pattern.compile(EMAIL_REGEX_PATTERN, Pattern.CASE_INSENSITIVE)
```

Comment:
Make the Pattern static, not the string.
It is expensive.

Reply:
updated